### DISCUSSION TITLE
Jaeger tracing integration problem

### DISCUSSION THREAD

**Discussion by claudiuavat1** [kiali/kiali#5899] [General]
Hi,

I have a very interesting problem that I couldn't find any thread for.

Traces from Jaeger are being presented just fine in the _Applications_ view, but not in the _Workloads_ view, where it is much more important.

Setup:
Kiali: v1.59.0
Service Mesh: Istio 1.15.2
Jaeger: all-in-one 1.42.0 (in memory)

Configuration (helm values):
external_services:
..tracing:
....auth:
......type: none
....enabled: true
....use_grpc: true
....namespace_selector: true
....in_cluster_url: 'http://tracing.istio-system.svc.cluster.local:16685/jaeger'
....url: 'https://<private_domain>/jaeger'

Research:

In _Applications_ I see the following api call successfully returning the expected results:
GET https://<private_domain>/kiali/api/namespaces/<my_ns>/**apps**/<my_app>/traces?...
{"data":[a lot], "errors":null,"jaegerServiceName":"<my_app>.<my_ns>"}

In _Workloads_ I see the following api call returning 0 results:
GET https://<private_domain>/kiali/api/namespaces/<my_ns>/**workloads**/<my_app>/traces?...
{"data":[], "errors":null,"jaegerServiceName":"<my_app>.<my_ns>"}
If i click on "View in Tracing" I do see the search in Jaeger populated properly and with the expected traces visualized in the map (Service: <my_app>.<my_ns>, Operation: all, Lookback: Last Hour)

Another weird one, if it can help isolate the problem, is that in _Services_ I also don't see any traces, but the return payload is a little bit weird:
GET https://<private_domain>/kiali/api/namespaces/<my_ns>/**services**/<my_app>/traces?...
{"data":[], "errors":null,"jaegerServiceName":"**.**<my_ns>"} (not showing <my_app> before dot)

I have the same name and labels on the Deployment and Pod definitions:
metadata:
  name: <my_app>
...
template:
   metadata:
     labels:
       app: <my_app>
       version: 'id'
       app.kubernetes.io/name: <my_app>
       app.kubernetes.io/instance: <my_app>
       app.kubernetes.io/version: 'id'

Any help will be greatly appreciated. What could it be?

**Comment by claudiuavat1** [comment]
I found what the problem is/was: not having the envoy proxy sending spans to Jaeger to enrich the trace data. 

I am still required to work with B3 headers and the old zipkin protocol. Will migrate soon to open telemetry.

**Comment by jshaughn** [comment]
Thanks for answering the question!

One more note is that the traces are queried at the "app" level, and so not every query response will apply to a specific workload.  In other words, for a workload, we figure out which "app" the workload is part of, query traces for that app, and then present the subset of traces that hit the workload.   This sometimes causes some confusion as sometimes less traces than expected are returned.  This isn't really your issue here, but I thought I'd mention it.
